### ISSUE TITLE
Torchtext datasets failed during multi-processing due to lambda functions usage

### ISSUE THREAD

**Issue by parmeet** [pytorch/text#1716]
Duplicate Issue https://github.com/pytorch/data/issues/397

# Issue
Currently, torchtext datasets make use of lambda functions that creates issue for MultiProcessing with DataLoader as pickle doesn't support serialization with lambda functions. 

Tracking the issue here as well for visibility 

# Suggested Resolution
Remove usage of lambda functions

cc: @NivekT  , @NicolasHug

**Comment by NicolasHug** [comment]
Thanks @parmmet for keeping track of this!

I think there are still instances of the use of `lambdas` in the tutorials:

https://github.com/pytorch/text/blob/dfb53af611a6d6dc0a043a6eb26a1cec30501fc1/examples/tutorials/sst2_classification_non_distributed.py#L98

The DataLoader doesn't use multi-processing for now 

https://github.com/pytorch/text/blob/dfb53af611a6d6dc0a043a6eb26a1cec30501fc1/examples/tutorials/sst2_classification_non_distributed.py#L101

so it shouldn't be a problem in the tutorial script itself.

But in general, users might start tweaking the tutorial and set `num_workers=k`. This might cause problems because of the use of `lambdas`, but also with the RNG and shuffling issues that I raised in https://github.com/pytorch/data/issues/302.

Since these datasets are released already, I was wondering if torchtext had a plan to document these pitfalls before the next release?

**Comment by parmeet** [comment]
Thanks @NicolasHug, sure will update the tutorial. 

Regarding some of the pitfalls, I think it's a good idea to document.  Will look into it and see to create a tracking issue that can be referenced in release notes or something :).
